Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -2370,15 +2373,29 @@ void InlineCacheCompiler::emitProxyObjectAccess(ProxyObjectAccessCase& accessCas
 
     jit.storeCell(baseGPR, calleeFrame.withOffset(virtualRegisterForArgumentIncludingThis(0).offset() * sizeof(Register)));
 
-    if (!stubInfo.hasConstantIdentifier) {
-        RELEASE_ASSERT(accessCase.identifier());
-        GPRReg propertyGPR = stubInfo.propertyGPR();
-        jit.storeCell(propertyGPR, calleeFrame.withOffset(virtualRegisterForArgumentIncludingThis(1).offset() * sizeof(Register)));
-    } else
+    if (stubInfo.hasConstantIdentifier)
         jit.storeTrustedValue(accessCase.identifier().cell(), calleeFrame.withOffset(virtualRegisterForArgumentIncludingThis(1).offset() * sizeof(Register)));
+    else if (accessCase.identifier())
+        jit.storeCell(stubInfo.propertyGPR(), calleeFrame.withOffset(virtualRegisterForArgumentIncludingThis(1).offset() * sizeof(Register)));
+    else {
+        ASSERT(accessCase.requiresInt32PropertyCheck());
+
+        jit.makeSpaceOnStackForCCall();
+        jit.setupArguments<decltype(operationInt32ToStringWithRadix10)>(CCallHelpers::TrustedImmPtr(&vm), stubInfo.propertyGPR());
+        jit.prepareCallOperation(vm);
+
+        auto operationCall = jit.call(OperationPtrTag);
+        jit.addLinkTask([=] (LinkBuffer& linkBuffer) {
+            linkBuffer.link<OperationPtrTag>(operationCall, operationInt32ToStringWithRadix10);
+        });
+
+        jit.reclaimSpaceOnStackForCCall();
+        jit.storeCell(GPRInfo::returnValueGPR, calleeFrame.withOffset(virtualRegisterForArgumentIncludingThis(1).offset() * sizeof(Register)));
+    }
 
     switch (accessCase.m_type) {
     case AccessCase::ProxyObjectLoad:
+    case AccessCase::IndexedProxyObjectLoad:
         jit.storeCell(thisGPR, calleeFrame.withOffset(virtualRegisterForArgumentIncludingThis(2).offset() * sizeof(Register)));
```

Comment:
Why is `thisGPR` still usable after calling `operationInt32ToStringWithRadix10` operation?

Reply:
🤔 I'm likely missing something very obvious, yet `operationInt32ToStringWithRadix10` operates on `propertyGPR` and utilizes `GPRInfo::returnValueGPR`. Can the latter clash in any way with

```cpp
GPRReg thisGPR = stubInfo.thisValueIsInExtraGPR() ? stubInfo.thisGPR() : baseGPR;
```

?

Reply:
Because `operationInt32ToStringWithRadix10` is C++ function, all caller-save registers can be clobbered. So, I think `thisGPR` (and other registers except for `returnValueGPR`) no longer holds the original expected value.